ir/shared/src/main/scala/org/scalajs/ir/Serializers.scala

```diff
@@ -983,10 +983,11 @@ object Serializers {
       }
     }
 
-    def writeOptHash(optHash: Option[TreeHash]): Unit = {
-      buffer.writeBoolean(optHash.isDefined)
-      for (hash <- optHash)
-        buffer.write(hash.hash)
+    def writeOptHash(version: Version): Unit = {
+      val isHash = version.isHash
+      buffer.writeBoolean(isHash)
+      if (isHash)
+        version.writeHash(buffer)
```

Comment:
What happens/should happen if the version is `Combined` here?

Reply:
Do not write it. The idea (and yes I'll document this properly) is that we'll only ever serialize hashed versions. All the other versions are not serialization safe (as in: we cannot guarantee collision freedom across linker instances).